upi/openstack/bootstrap.yaml

```
@@ -40,3 +40,4 @@
       state: present
       network: "{{ os_external_network }}"
       server: "{{ os_bootstrap_server_name }}"
+    when: os_api_fip is defined and os_api_fip|length>0
```

It's better to check if `os_external_network` was set, as it's a prerequisite for creating the FIP:
```suggestion
    when: os_external_network is defined and os_external_network|length>0
```

---

What if the user sets the external network, but not the FIP? I found it interesting to use FIPs if passed, and not use them if not passed.

Maybe I could clarify in `inventory.yaml` that you should set an external network if you set the FIPs?

---

> What if the user sets the external network, but not the FIP?

I'd argue this has nothing to do with the bootstrap node. And yes, this is likely going to break, unless they also modify the scripts to use an existing network configured with a router to an external network, and the FIP belong to the CIDR of the external subnet. But that's beyond the point.

For IPI, we do not allow setting a FIP without also setting an external network. Or at least that was the intent, https://github.com/openshift/installer/pull/3822#discussion_r454903882

---

makes perfect sense :+1: thanks